### DISCUSSION TITLE
ESPHome Support

### DISCUSSION THREAD

**Discussion by stroodle96** [microphonon/acoustic-leak-detector#5] [Ideas]
Hello,
This sensor seems like a perfect match with esphome to get the sensor results in home assistant easily. ESPHome supports I2C sensors. Could someone put together an example config yaml for esphome that makes it easy to get started with this sensor and home assistant?

**Comment by microphonon** [comment]
This should not be too difficult. I don't have any Home Assistant hardware or experience with esphome, so I am not presently in a position to do any development for it. There is code for the ThingSpeak platform (ESP8266) in this repository, so maybe that could be used as a starting point. I have attempted to make the I2C command set (read and write) as simple as possible.

**Comment by microphonon** [comment]
The AquaPing leak sensor is now working with ESPHome. I got it implemented with the D1-mini ESP8266 module that uses WiFi to communicate with HA. For development I ran the HA operating system in a VirtualBox on Ubuntu, but it should be agnostic to the host hardware. Relevant files (firmware, yaml, dashboard) have been uploaded to this repository. I will expand the documentation next, but wanted to provide a starting point for anyone who may be interested.

**Reply by microphonon** [reply]
[https://github.com/microphonon/acoustic-leak-detector/tree/main/Home_Assistant](https://github.com/microphonon/acoustic-leak-detector/tree/main/Home_Assistant)
